backend/public-graph/graph/resolver.go

```diff
@@ -1724,7 +1724,7 @@ func (r *Resolver) sendErrorAlert(ctx context.Context, projectID int, sessionObj
 			}
 			excluded := false
 			for _, env := range excludedEnvironments {
-				if env != nil && *env == sessionObj.Environment {
+				if env != nil && *env == errorObject.Environment {
```

Comment:
@Vadman97 @mayberryzane does this change make sense to you? I think this resolves one of the issues discussed in [this Slack thread](https://highlightcorp.slack.com/archives/C062LLVR9LK/p1704963364911419), but wanted to confirm there wasn't a reason to be comparing against the session's environment rather than the error's. Assuming it's just because we didn't support environment on errors when the error alerts were implemented, but thought I'd double check. Thanks!

Reply:
yup, that's exactly right. a frontend error's environment [will be set](https://github.com/highlight/highlight/blob/da48ded61786ebaccb54d74e59345a43beb1abcc/backend/public-graph/graph/resolver.go#L2744) to the session's environment anyways, and for backend errors using the backend sdk's reported environment makes much more sense